src/save/save_slot.rs

```diff
@@ -1496,14 +1580,23 @@ impl Default for SaveSlot {
 }
 
 impl Read for SaveSlot {
-    fn read(br: &mut BinaryReader) -> Result<SaveSlot, io::Error> {
+    fn read_pc(br: &mut BinaryReader) -> Result<SaveSlot, io::Error> {
         let mut save_slot = SaveSlot::default();
 
         // Slot end position
         let end = br.pos + 0x280010;
 
         // Checksum
-        save_slot.checksum.copy_from_slice(br.read_bytes(0x10)?);
+        let checksum_bytes: &[u8] = br.read_bytes(0x10)?;
```

Comment:
I'm unsure if there is a better implementation now that the checksum_array is an Option. read_bytes is apparently &[u8], but I don't know if it needs to be 16 bytes. Let me know.